Jenkinsfile

```diff
@@ -0,0 +1,83 @@
+#!/usr/bin/env groovy
+
+pipeline {
+  agent {
+    label 'huge && bionic && x86_64 && !xenial && !trusty'
+  }
+
+  environment {
+    NPROC = "0"
+  }
+
+  stages {
+    stage('Prepare') {
+      steps {
+        sh '''#!/usr/bin/env bash
+        set -exo pipefail
+        source ./run-make-check.sh
+        FOR_MAKE_CHECK=1 prepare
+        '''
+      }
+    }
+
+    stage('Build') {
+      steps {
+        sh '''#!/usr/bin/env bash
+        set -exo pipefail
+        source ./run-make-check.sh
+        configure
+        build tests
+        '''
+      }
+    }
+
+    stage('Test') {
+      parallel {
+        stage('Unit Tests') {
+          steps {
+            sh '''#!/usr/bin/env bash
+            set -exo pipefail
+            source ./run-make-check.sh
+            cd build
+            export CHECK_MAKEOPTS="$DEFAULT_MAKEOPTS --no-compress-output -T Test"
+            run
+            '''
+          }
+          post {
+            always{
+              xunit(
+                [CTest(
+                  deleteOutputFiles: true,
+                  failIfNotNew: true,
+                  pattern: 'build/Testing/**/Test.xml',
+                  skipNoTestFiles: true,
+                  stopProcessingIfError: true)
+                ]
+              )
+            }
+          }
+        }
+
+        stage('API Tests') {
+          // Disabled so far
+          when { expression { false } }
+
+          steps {
+            sh '''#!/usr/bin/env bash
+            set -exo pipefail
+            cd src/pybind/mgr/dashboard
+            ./install-backend-api-test-deps.sh
+            '''
+
+            sh '''#!/usr/bin/env bash
+            set -exo pipefail
+            cd src/pybind/mgr/dashboard
+            ./run-backend-api-tests.sh
+            '''
+          }
+        }
+      } // parallel
```

Comment:
the lint-alike tests can be integrated into the pipeline like
```groovy
stage('Lint') {
  // assuming we define a target of "lint-tests" which groups all lint tests,
  // and one of them writes the output to "out/flake8.txt"
  sh "make lint-tests"
  def flake8_issues = scanForIssues tool: flake8(pattern:'out/flake8.txt')
  publishIssues id: 'flake8-ceph', name: 'flake8-ceph', issues: [flake8_issues]
}
```

Reply:
I was thinking on relying on ctest labels (perhaps relying on https://github.com/ceph/ceph/pull/29844) for only running linting tests. How does that sound to you?

Reply:
i think #29844 is orthogonal to this. but yes, as long as these lint tests are not added to the "tests" target, i am for it.

Reply:
FWIW, i have https://jenkins.ceph.com/blue/organizations/jenkins/cbt/detail/wip-jenkins-pipeline/10/pipeline/ when testing the GitHub Checks plugin.